sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -429,6 +434,85 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
         }).map(ar -> new MsalToken(ar, options));
     }
 
+    /**
+     * Asynchronously acquire a token from Active Directory with a device code challenge. Active Directory will provide
+     * a device code for login and the user must meet the challenge by authenticating in a browser on the current or a
+     * different device.
+     *
+     * @param request the details of the token request
+     * @return a Publisher that emits an AccessToken when the device challenge is met, or an exception if the device
+     *     code expires
+     */
+    public Mono<MsalToken> authenticateWithVsCodeCredential(TokenRequestContext request) {
+
+        VisualStudioCacheAccessor accessor = new VisualStudioCacheAccessor();
+
+        JsonNode userSettings = accessor.getUserSettings();
+
+        String tenant = tenantId;
+        String cloud = "Azure";
+
+        if (!userSettings.isNull()) {
+            if (userSettings.has("azure.tenant")) {
+                tenant = userSettings.get("azure.tenant").asText();
+            }
+
+            if (userSettings.has("azure.cloud")) {
+                cloud = userSettings.get("azure.cloud").asText();
+            }
+        }
+
+        String credential = accessor.getCredentials("VS Code Azure", cloud);
+
+        String authority = KnownAuthorityHosts.AZURE_CLOUD.replaceAll("/+$", "")
+            + "/organizations/" + tenant;
```

Comment:
The authority has to be configurable why is this hard-coded to public cloud? Also I think it should append EITHER '/organizations/' OR tenantId. At least that's what we do in other languages. Have you confirmed that /orginizations/<tenantId> actually works?

Reply:
To be more precise here the authority should be determined by the azure.cloud vscode setting, and the tenantId should use whatever was passed in through code, or if not specified default to the vscode setting.

Reply:
Yep, added the logic to support multiple clouds in both VScode + IntelliJ creds.

Reply:
/organizations/ + tenantid 
works well.
tenantId is either user passed or "common"
Should I follow Either strategy ? to align with other languages ?